src/main/java/org/jabref/logic/importer/fileformat/CffImporter.java

```diff
@@ -175,27 +287,97 @@ public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
         fieldMappings.put("license", BiblatexSoftwareField.LICENSE);
+        fieldMappings.put("message", StandardField.COMMENT);
         fieldMappings.put("repository", BiblatexSoftwareField.REPOSITORY);
+        fieldMappings.put("title", StandardField.TITLE);
         fieldMappings.put("url", StandardField.URL);
-        fieldMappings.put("abstract", StandardField.ABSTRACT);
-        fieldMappings.put("message", StandardField.COMMENT);
-        fieldMappings.put("date-released", StandardField.DATE);
-        fieldMappings.put("keywords", StandardField.KEYWORDS);
+        fieldMappings.put("version", StandardField.VERSION);
+
+        // specific to references and preferred-citation
+        fieldMappings.put("edition", StandardField.EDITION);
+        fieldMappings.put("isbn", StandardField.ISBN);
+        fieldMappings.put("issn", StandardField.ISSN);
+        fieldMappings.put("issue", StandardField.ISSUE);
+        fieldMappings.put("journal", StandardField.JOURNAL);
+        fieldMappings.put("month", StandardField.MONTH);
+        fieldMappings.put("notes", StandardField.NOTE);
+        fieldMappings.put("number", StandardField.NUMBER);
+        fieldMappings.put("pages", StandardField.PAGES);
+        fieldMappings.put("status", StandardField.PUBSTATE);
+        fieldMappings.put("volume", StandardField.VOLUME);
+        fieldMappings.put("year", StandardField.YEAR);
         return fieldMappings;
     }
 
     private List<String> getUnmappedFields() {
         List<String> fields = new ArrayList<>();
 
+        fields.add("abbreviation");
+        fields.add("collection-doi");
+        fields.add("collection-title");
+        fields.add("collection-type");
         fields.add("commit");
+        fields.add("copyright");
+        fields.add("data-type");
+        fields.add("database");
+        fields.add("date-accessed");
+        fields.add("date-downloaded");
+        fields.add("date-published");
+        fields.add("department");
+        fields.add("end");
+        fields.add("entry");
+        fields.add("filename");
+        fields.add("format");
+        fields.add("issue-date");
+        fields.add("issue-title");
         fields.add("license-url");
-        fields.add("repository-code");
+        fields.add("loc-end");
+        fields.add("loc-start");
+        fields.add("medium");
+        fields.add("nihmsid");
+        fields.add("number-volumes");
+        fields.add("patent-states");
+        fields.add("pmcid");
         fields.add("repository-artifact");
-
+        fields.add("repository-code");
+        fields.add("scope");
+        fields.add("section");
+        fields.add("start");
+        fields.add("term");
+        fields.add("thesis-type");
+        fields.add("volume-title");
+        fields.add("year-original");
         return fields;
     }
+
+    private HashMap<String, EntryType> getTypeMappings() {
+        HashMap<String, EntryType> typeMappings = new HashMap<>();
```

Comment:
Make map `public static final` to avoid initialization at each import.

(This also removes the method)

---

Doesn't share the export and import the same mappings? Shouldn't a `public static final` [HashBiMap](https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/HashBiMap.java) be shared among the importer and exporter?

Reply:
You're right, I wrote mappings in the exporter and I import them in the importer (no pun intended  :)), using `HashBiMap`. I did not know about these !